docs/storage/commit_log/commit_log_based_storage_design.md

```diff
@@ -0,0 +1,354 @@
+## Commit-Log based Trillian storage
+
+*Status: Draft*
+
+*Authors: al@google.com, drysdale@google.com, filippo@cloudflare.com*
+
+*Last Updated: 2017-05-11*
+
+
+[TOC]
+
+
+## Objective
+
+A design for an alternative Trillian storage layer which uses a distributed and
+immutable *commit log* as the source of truth for a Trillian Log's contents and
+sequence information, and one or more independent *"readonly"* databases built
+from the commit log to serve queries.
+
+This design allows for:
+
+*   flexibility in scaling Trillian deployments,
+*   easier recovery from corrupt/failed database deployments since operators
+can simply delete it and allow it to rebuild itself from the commit log.
```

Comment:
Not without an operational impact though presumably? One does not simply delete a database in Mordor.